### PR TITLE
improve how branch images skips

### PR DESCRIPTION
**Pull request by jakecoffman**
It's difficult to tell if the branch images job was skipped or not, making it hard to pick the SHA out of the output and do a deployment.

This moves the part where it figures out if the PR is approved to a first step, and if it's not approved it skips the entire `push-updater-images` which makes it really obvious.

### CODE REVIEW

**Comment by jakecoffman** [comment]
Don't approve this yet, still working out the kinks.

**Comment by jakecoffman** [comment]
That's crystal clear.

Screenshot 2023-12-19 at 3 09 32 PM

Ready for approval.

File: .github/workflows/images-branch.yml [hunk 1]
```diff
@@ -18,9 +18,37 @@ on:  # yamllint disable-line rule:truthy
...
+      # Skips push-updater-images by setting an output
+      - name: Check if pull request is approved
+        run: |
+          DECISION=$(gh pr view ${{ env.PR }} --json reviewDecision,state -t '{{.reviewDecision}}:{{.state}}')
+          echo "decision=$DECISION" >> $GITHUB_OUTPUT
+
+      # overwrite the previous result on pull request events on forks since forks can't publish to GHCR
+      - name: Skip forks
+        if: github.event.pull_request.head.repo.fork
+        run: echo "DECISION=FORK" >> $GITHUB_OUTPUT
```
Comment (Nishnha) [inline .github/workflows/images-branch.yml:46]: does it matter that DECISION is in caps here but it's lowercase when referenced?

Comment (jakecoffman) [reply .github/workflows/images-branch.yml]: Fixed, nice catch!

**Review by Nishnha — APPROVED** [review]
LGTM!

File: .github/workflows/images-branch.yml [hunk 2]
```diff
@@ -18,9 +18,35 @@ on:  # yamllint disable-line rule:truthy
...
+        id: decision
+        run: |
+          DECISION=$(gh pr view ${{ env.PR }} --json reviewDecision,state -t '{{.reviewDecision}}:{{.state}}')
+          echo "decision=$DECISION" >> $GITHUB_OUTPUT
+
   push-updater-images:
-    name: Deploy
     runs-on: ubuntu-latest
+    needs: approval
+    if: needs.approval.outputs.decision == 'APPROVED:OPEN' && !github.event.pull_request.head.repo.fork
```
Comment (jeffwidman) [inline .github/workflows/images-branch.yml:49]: I found it very confusing that we checked for approval & also checked that not a fork, so once I puzzled through to understand why this was written this way, I was able to tidy it up a little more:
* https://github.com/dependabot/dependabot-core/pull/11212

**Pull request merged**